crabs/tracker/track_video.py

```diff
@@ -171,52 +159,58 @@ def run_tracking(self):
             )
             return
 
-        # In any case run inference
         # initialisation
-        frame_number = 1
-        self.tracked_list = []
+        frame_idx = 0
+        self.tracked_bbox_id = []
 
         # Loop through frames of the video in batches
         while self.video.isOpened():
             # Break if beyond end frame (mostly for debugging)
             if (
                 self.args.max_frames_to_read
-                and frame_number > self.args.max_frames_to_read
+                and frame_idx > self.args.max_frames_to_read
```

Comment:
```suggestion
                and frame_idx + 1 > self.args.max_frames_to_read
```
or 
```suggestion
                and frame_idx >= self.args.max_frames_to_read
```
Since `self.args.max_frames_to_read` is not 0-based right?

E.g. If we want to read 3 frames max, then `self.args.max_frames_to_read = 3`. But when 3 frames have been read, the index at the start of the 4th iteration is `frame_idx=3`, which is not larger than `self.args.max_frames_to_read`. So we would read a fourth frame, right?